src/azul/service/manifest_service.py

```diff
@@ -776,6 +809,409 @@ def manifest_config(self) -> ManifestConfig:
 Bundles = MutableMapping[FQID, Bundle]
 
 
+@dataclass(frozen=True)
```

Comment:
We are migrating to `attr.s` with auto_attribs=True, kwonly=True and frozen=True. New code should use that out of the box.